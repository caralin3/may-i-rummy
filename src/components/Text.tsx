import React from 'react';
import { Text as RNText, TextStyle } from 'react-native';
import { FontSizes, GutterSizes } from '../appearance';

export interface TextProps {
  align?: TextStyle['textAlign'];
  bold?: boolean;
  gutters?: {
    bottom?: keyof typeof GutterSizes;
    left?: keyof typeof GutterSizes;
    right?: keyof typeof GutterSizes;
    top?: keyof typeof GutterSizes;
  };
  italic?: boolean;
  size?: keyof typeof FontSizes;
  style?: TextStyle;
  // textColor?: keyof typeof defaultTheme.colors;
}

export const Text: React.FC<TextProps> = ({
  align,
  bold,
  children,
  // textColor,
  gutters,
  italic,
  size = 'md',
  style,
}) => (
  <RNText
    style={{
      ...style,
      // color: textColor
      //   ? defaultTheme.colors[textColor].toString()
      //   : !!!!style && !!style.color
      //   ? style.color
      //   : undefined,
      fontSize: FontSizes[size],
      fontStyle: italic ? 'italic' : undefined,
      fontWeight: bold ? 'bold' : undefined,
      lineHeight: 24,
      paddingBottom: !!gutters && !!gutters.bottom ? GutterSizes[gutters.bottom] : undefined,
      paddingLeft: !!gutters && !!gutters.left ? GutterSizes[gutters.left] : undefined,
      paddingRight: !!gutters && !!gutters.right ? GutterSizes[gutters.right] : undefined,
      paddingTop: !!gutters && !!gutters.top ? GutterSizes[gutters.top] : undefined,
      textAlign: align ? align : undefined,
    }}
  >
    {children}
  </RNText>
);
