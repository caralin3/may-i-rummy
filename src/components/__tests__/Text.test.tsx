import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from '../Text';

describe('Text', () => {
  test('renders correctly', () => {
    const wrapper = renderer.create(<Text>Text</Text>).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  test('renders bold correctly', () => {
    const wrapper = renderer.create(<Text bold>Text</Text>).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  test('renders italic correctly', () => {
    const wrapper = renderer.create(<Text italic>Text</Text>).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  test('renders large font size correctly', () => {
    const wrapper = renderer.create(<Text size='lg'>Text</Text>).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
