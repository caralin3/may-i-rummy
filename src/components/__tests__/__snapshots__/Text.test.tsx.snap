// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders bold correctly 1`] = `
<Text
  style={
    Object {
      "color": undefined,
      "fontSize": 24,
      "fontStyle": undefined,
      "fontWeight": "bold",
      "lineHeight": 24,
      "paddingBottom": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "textAlign": undefined,
    }
  }
>
  Text
</Text>
`;

exports[`Text renders correctly 1`] = `
<Text
  style={
    Object {
      "color": undefined,
      "fontSize": 24,
      "fontStyle": undefined,
      "fontWeight": undefined,
      "lineHeight": 24,
      "paddingBottom": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "textAlign": undefined,
    }
  }
>
  Text
</Text>
`;

exports[`Text renders italic correctly 1`] = `
<Text
  style={
    Object {
      "color": undefined,
      "fontSize": 24,
      "fontStyle": "italic",
      "fontWeight": undefined,
      "lineHeight": 24,
      "paddingBottom": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "textAlign": undefined,
    }
  }
>
  Text
</Text>
`;

exports[`Text renders large font size correctly 1`] = `
<Text
  style={
    Object {
      "color": undefined,
      "fontSize": 32,
      "fontStyle": undefined,
      "fontWeight": undefined,
      "lineHeight": 32,
      "paddingBottom": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "textAlign": undefined,
    }
  }
>
  Text
</Text>
`;
